src/cpp/fastdds/xtypes/serializers/json/dynamic_data_json.cpp

```diff
@@ -991,6 +880,209 @@ ReturnCode_t json_serialize_array(
     return ret;
 }
 
+ReturnCode_t json_serialize_map_member(
+        const traits<DynamicDataImpl>::ref_type& data,
+        const std::string& member_name,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    ReturnCode_t ret = RETCODE_OK;
+    nlohmann::json j_map;
+    const TypeDescriptorImpl& map_desc = data->enclosing_type()->get_descriptor();
+    traits<DynamicTypeImpl>::ref_type key_type = traits<DynamicType>::narrow<DynamicTypeImpl>(
+        map_desc.key_element_type())->resolve_alias_enclosed_type();
+    traits<DynamicTypeImpl>::ref_type value_type = traits<DynamicType>::narrow<DynamicTypeImpl>(
+        map_desc.element_type())->resolve_alias_enclosed_type();
+
+    std::map<std::string, MemberId> key_to_id;
+    if (RETCODE_OK != (ret = data->get_keys(key_to_id)))
+    {
+        EPROSIMA_LOG_ERROR(XTYPES_UTILS,
+                "Error encountered while serializing map member to JSON: get_keys failed.");
+        return ret;
+    }
+
+    std::map<MemberId, std::string> id_to_key;
+    for (const auto& it : key_to_id)
+    {
+        id_to_key[it.second] = it.first;
+    }
+    assert(id_to_key.size() == key_to_id.size());
+
+    uint32_t size = data->get_item_count();
+    assert(size == key_to_id.size());
+    for (uint32_t i = 0; i < size; ++i)
+    {
+        MemberId id = data->get_member_id_at_index(i);
+        if (MEMBER_ID_INVALID == id)
+        {
+            EPROSIMA_LOG_ERROR(XTYPES_UTILS,
+                    "Error encountered while serializing map member's member to JSON: invalid member id.");
+            return RETCODE_BAD_PARAMETER;
+        }
+
+        auto it = id_to_key.find(id);
+        if (it == id_to_key.end())
+        {
+            EPROSIMA_LOG_ERROR(XTYPES_UTILS,
+                    "Error encountered while serializing map member's member to JSON: key not found.");
```

Comment:
As `id_to_key` is only used to find the id, perhaps we could just reuse the `key_to_id` map:
```suggestion
        auto it = std::find_if(key_to_id.begin(), key_to_id.end(), [&id](const auto& value){value.second == id});
```

Reply:
Yes, you're right. Closing this